artiq/examples/master/zotino.py

```diff
@@ -0,0 +1,41 @@
+from artiq.experiment import *
+
+class ZotinoTestDAC(EnvExperiment):
+    def build(self):
+        self.setattr_device("core")
+        self.setattr_device("dac_zotino")
+        self.setattr_device("latch_config")
+        self.setattr_device("clk_config")
+        self.setattr_device("ser_config")
+		
+    @kernel
+    def dir_config(self, rclk, srclk, ser, data): 
+        dt = 10*us
+        srclk.off()
+        rclk.off()
+        delay(dt)
+        x = 0x80000000
+        for i in range(32):  
```

Comment:
There are other shift registers in the system, e.g. for the Novogorny PGIA, with other lengths. It makes sense to make this function generic to transfer any number of bits (not just 32), document it, and move it into some library (e.g. ``artiq.coredevice.shiftreg``).

Reply:
And this function could be turned into a device that uses and depends on the TTL devices, just like the DAC driver that depends on a SPI device for the data and a TTL device for LDAC.